src/core/public/core_system.ts

```diff
@@ -85,15 +94,18 @@ export class CoreSystem {
       },
     });
 
-    this.notificationsTargetDomElement = document.createElement('div');
+    this.notificationsTargetDomElement$ = new Subject();
     this.notifications = new NotificationsService({
-      targetDomElement: this.notificationsTargetDomElement,
+      targetDomElement$: this.notificationsTargetDomElement$.asObservable(),
```

Comment:
Why make this change in this PR? It seems unrelated.

Reply:
Now that `LegacyService#setup` is async this was necessary to ensure that the `NotificationsService` does not attempt to render toasts before this DOM element is mounted on the page. Without this change, it's possible a toast that was added by a plugin's `setup` may get cleared (by the toast timeout) before it was even actually shown to the user.